File: 804-unique-morse-code-words/804-unique-morse-code-words.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        c = Counter()
        res = 0
        f = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        
        for word in words:
            code = ""
            for w in word:
                code += f[(ord(w)-97)]
            
            c[code] += 1
        
        return len(c.values())
                
        
        

File: 804-unique-morse-code-words/804-unique-morse-code-words.java
class Solution {
    public int uniqueMorseRepresentations(String[] words) {
        String[] d = new String[] {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        Set<String> set = new HashSet<>();
        for (String word : words){
            StringBuilder sb = new StringBuilder();
            for (int i = 0; i < word.length(); i++){
                sb.append(d[word.charAt(i) - 'a']);
            }
            set.add(sb.toString());
        }
        return set.size();
    }
}